### PR TITLE
Split interop into interop-cats,interop-future and interop-scalaz7x.

### PR DESCRIPTION
**Pull request by hepin1989**
refs:https://github.com/scalaz/scalaz-zio/issues/394

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -84,26 +98,64 @@ lazy val interop = crossProject(JSPlatform, JVMPlatform)
   .in(file("interop"))
   .settings(stdSettings("zio-interop"))
   .dependsOn(core % "test->test;compile->compile")
+  .settings(
+    scalacOptions in Test ++= Seq("-Yrangepos")
+  )
+
+lazy val interopJVM = interop.jvm
+
+lazy val interopJS = interop.js
```
Comment (hepin1989) [inline build.sbt]: Should they be renamed to `interopSharedJVM` and `interopSharedJS`?

Comment (jdegoes) [reply build.sbt]: Yes, probably, and placed in `interop-shared`.

**Comment by jdegoes** [comment]
@hepin1989 One more change and should be good to go: can you update the microsite so it contains appropriate directions for each of the different modules? They will be different JARs now, and will require separate changes to the sbt build file dependencies.

**Comment by hepin1989** [comment]
@jdegoes I have updated it.

**Comment by jdegoes** [comment]
@hepin1989 Perfect! If you can resolve the few conflicts, we'll get this merged. Thanks for your work on this!

**Comment by hepin1989** [comment]
@jdegoes Let me first fix my Akka PR:)

**Comment by hepin1989** [comment]
@jdegoes I have rebased, it's your turn now:)

**Pull request merged**